contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,239 @@
+            "Caller is not the Strategist"
+        );
+        _;
+    }
+
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _vaultAddress Address of the Vault
+    /// @param _beaconChainDepositContract Address of the beacon chain deposit contract
+    /// @param _ssvNetwork Address of the SSV Network contract
+    constructor(
+        address _wethAddress,
+        address _vaultAddress,
+        address _beaconChainDepositContract,
+        address _ssvNetwork
+    ) {
+        WETH_TOKEN_ADDRESS = _wethAddress;
+        BEACON_CHAIN_DEPOSIT_CONTRACT = _beaconChainDepositContract;
+        SSV_NETWORK_ADDRESS = _ssvNetwork;
+        VAULT_ADDRESS = _vaultAddress;
+    }
+
+    /// @notice Set the address of the registrator which can register, exit and remove validators
+    function setRegistrator(address _address) external onlyGovernor {
+        emit RegistratorChanged(_address);
+        validatorRegistrator = _address;
+    }
+
+    /// @notice Stakes WETH to the node validators
+    /// @param validators A list of validator data needed to stake.
+    /// The `ValidatorStakeData` struct contains the pubkey, signature and depositDataRoot.
+    /// Only the registrator can call this function.
+    function stakeEth(ValidatorStakeData[] calldata validators)
+        external
+        onlyRegistrator
+        whenNotPaused
+    {
+        uint256 requiredETH = validators.length * 32 ether;
+
+        // Check there is enough WETH from the deposits sitting in this strategy contract
+        require(
+            requiredETH <= IWETH9(WETH_TOKEN_ADDRESS).balanceOf(address(this)),
+            "insufficient WETH"
+        );
+
+        // Convert required ETH from WETH
+        IWETH9(WETH_TOKEN_ADDRESS).withdraw(requiredETH);
+
+        // For each validator
+        for (uint256 i = 0; i < validators.length; ) {
+            bytes32 pubkeyHash = keccak256(validators[i].pubkey);
+            VALIDATOR_STATE currentState = validatorsStates[pubkeyHash];
+
+            require(
+                currentState == VALIDATOR_STATE.REGISTERED,
+                "Validator not registered"
+            );
+
+            /* 0x01 to indicate that withdrawal credentials will contain an EOA address that the sweeping function
+             * can sweep funds to.
+             * bytes11(0) to fill up the required zeros
+             * remaining bytes20 are for the address
+             */
+            bytes memory withdrawal_credentials = abi.encodePacked(
+                bytes1(0x01),
+                bytes11(0),
+                address(this)
+            );
+            IDepositContract(BEACON_CHAIN_DEPOSIT_CONTRACT).deposit{
+                value: 32 ether
+            }(
+                validators[i].pubkey,
+                withdrawal_credentials,
+                validators[i].signature,
+                validators[i].depositDataRoot
+            );
+
+            activeDepositedValidators += 1;
```

Comment:
You could save on some gas by adding the `validators.length` once instead of doing a +1 on every interaction..

Reply:
great gas savings thanks: https://github.com/OriginProtocol/origin-dollar/pull/2053/commits/fd4f2d24d5db8723e7eef9ed8630a4f221684749